examples/javascript/groveeldriver.js

```diff
@@ -0,0 +1,54 @@
+/*jslint node:true, vars:true, bitwise:true, unparam:true */
+/*jshint unused:true */
+/*
+* Author: Zion Orent <zorent@ics.com>
+* Copyright (c) 2015 Intel Corporation.
+*
+* Permission is hereby granted, free of charge, to any person obtaining
+* a copy of this software and associated documentation files (the
+* "Software"), to deal in the Software without restriction, including
+* without limitation the rights to use, copy, modify, merge, publish,
+* distribute, sublicense, and/or sell copies of the Software, and to
+* permit persons to whom the Software is furnished to do so, subject to
+* the following conditions:
+*
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+* NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+* LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+* OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+* WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+var eldriver_lib = require("jsupm_groveeldriver");
+
+// The was tested with the Grove El Driver Module
+// Instantiate a Grove El Driver on digital pin D2
+var eldriver_obj = new eldriver_lib.GroveElDriver(2);
+
+var lightState = true;
+
+var myInterval = setInterval(function()
+{
+	if (lightState)
+		eldriver_obj.on();
+	else
+		eldriver_obj.off();
+	lightState = !lightState;
+}, 1000);
+
+// When exiting: turn off EL wire, clear interval, and print message
+process.on('SIGINT', function()
+{
+	clearInterval(myInterval);
+	eldriver_obj.off();
+	eldriver_obj = null;
+	eldriver_lib.cleanUp();
+	eldriver_lib = null;
+	console.log("Exiting...");
+	process.exit(0);
```

Comment:
Unfortunately, v8 doesn't guarantee garbage collection.
https://code.google.com/p/v8-juice/wiki/CommentaryOnV8#No_GC_guaranty

In testing upm, we found that GC wasn't called-- neither on scope exit nor on program exit. This is why we call it manually in the examples.

I wouldn't make a 1-to-1 comparison between JavaScript behavior in a browser and JavaScript behavior in nodeJS. Pretty much all browsers perform correct GC, and browser JavaScript likely hasn't required any type of variable destruction for quite a while.

But nodeJS is the wild west regarding GC. As of node 0.10.0, any attempt at guessing when to collect garbage was removed. This was because nodeJS couldn't figure out how to call GC in an efficient manner (see "Latency and Idle Garbage Collection" in the link).
http://blog.nodejs.org/2013/03/11/node-v0-10-0-stable/
This leaves us with V8 GC-- which isn't guaranteed.

Essentially, v8 was built by Google for Google Chrome. Everybody else using v8 has to deal with issues like this one from time to time.